interp/interp_test.go

```diff
@@ -840,6 +841,13 @@ BEGIN { x[1]=3; f5(x); print x[1] }
 	{`BEGIN { print >"out"; close("out"); getline <"out"; print >"out" }  # !awk !gawk`, "", "", "can't write to reader stream", ""},
 	{`BEGIN { print >"out"; close("out"); getline <"out"; print |"out" }  # !awk !gawk`, "", "", "can't write to reader stream", ""},
 
+	// The value of close() on a pipe emulates gawk behavior. Results are identical for both
+	// input and output. Windows does not do POSIX signals.
+	{`BEGIN { cmd="exit 9"; print "" |cmd; print close(cmd) } # !awk !posix`, "", "9\n", "", ""},
+	{`BEGIN { cmd="exec kill -9 $$"; print "" |cmd; print close(cmd) } # !awk !posix !windows-gawk`, "", "265\n", "", ""},
+	{`BEGIN { cmd="exit 9"; cmd |getline; print close(cmd) } # !awk !posix`, "", "9\n", "", ""},
+	{`BEGIN { cmd="exec kill -9 $$"; cmd |getline; print close(cmd) } # !awk !posix !windows-gawk`, "", "265\n", "", ""},
+
```

Comment:
The Windows tests are currently failing. I think we'll need to skip the `kill -9` ones on Windows (not just `windows-gawk`). Perhaps add `!windows` and then add this to the `TestInterp` runner:

```go
if strings.Contains(test.src, "!windows") && runtime.GOOS == "windows" {
    t.Skip("skipping on windows")
}
```

Reply:
Ok.

It looks like the cygwin signal emulation (or maybe mingw?) is automatically changing the exit status to `signal << 8`. A pity really. Because there technically are no signals in Windows, I believe the `syscall.WaitStatuc.Signal()` method will never return `true`.

See: `GOOS=windows go doc -src syscall.WaitStatus.Signaled`.

Reply:
I've added the `!windows` flag but I assume it means that the _goawk_ test should be skipped on Windows and not that all tests should be skipped on Windows. Otherwise the `!windows-gawk` flag becomes redundant and also, checking for `!windows` with contains will return true for `!windows-gawk` as well.

Because `strings.Contains(test.src, "!windows")` will also match `!windows-gawk` erronously (I assume), I have implemented a simple tokenizer that will check for flags after the comment and split them up, storing them into a map of bools, etc.

This should prove something interesting which I found when I dusted off my windows box and run these: gawk successfully returns the proper `close()` value when a child process is killed via signal. I suspect this is because the chocolatey gawk is compiled (statically) with cygwin. This is hard to prove as the package repo only hosts binaries. IOW the gawk test will pass... but not with --posix ofc ;-).

I also found and fixed a small number of tests which had `!gawk` flags but no comment char.

Reply:
According to this: https://github.com/benhoyt/goawk/actions/runs/6213666312/job/16864921268?pr=206 ...

_gawk_ from _chocolatey_ is running `cmd.exe` and not `sh` like you do for _goawk_ on Windows.

When I run the tests with _cygwin_ on Windows, where _gawk_ is installed by _cygwin_, these tests pass. That means _gawk_ on _cygwin_ is running piped commands in `sh` and not `cmd.exe`. (Other tests fail though.)

I suppose I should just skip all the signal tests on Windows by using `!windows !windows-gawk`?

Honestly this compatibility testing is a bit of a pain. I can see it as being useful in a generated report but embedding it into the CI as a go test for your own implementation is asking for trouble.

Reply:
I've added `!windows-gawk` and replaced `kill` with `/bin/kill` since it is more accurate, if we are not testing Windows anyways.

Reply:
No, I was definitely thinking `!windows` should mean skip this test on Windows, regardless of which interpreter (goawk, gawk, gawk --posix). I think `!windows !windows-gawk` is redundant and odd. So let's change `!windows` so it applies to both.

So this will mean we can skip all the signal tests on Windows by using just `!windows`.

The little tokenizer is a nice addition, though -- thanks!

Regarding this compatibility testing being a bit of pain -- you're not wrong, it's a bit messy. However, I think cleaning it up further is outside the scope of this PR, so let's just leave it for now.